contracts/contracts/levels/DoubleEntryPoint.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+interface DelegateERC20 {
+  function delegateTransfer(address to, uint256 value, address origSender) external returns (bool);
+}
+
+interface IAgent {
+    function handleTransaction(address user) external returns(bool);
+}
+
+interface IForta {
+    function setAgent(address agentAddress) external;
+    function notify(address user) external returns(bool);
+    function raiseAlert(address user) external;
+}
+
+contract Forta is IForta {
+  mapping(address => IAgent) public usersAgent;
+  mapping(address => uint256) public agentRaisedAlerts;
+
+  function setAgent(address agentAddress) external override {
+      require(address(usersAgent[msg.sender]) == address(0), "Agent already set");
+      usersAgent[msg.sender] = IAgent(agentAddress);
+  }
+
+  function notify(address user) external override returns(bool) {
+    require(address(usersAgent[user]) != address(0), "Can't find a suitable agent for user");
+    require(usersAgent[user].handleTransaction(user), "Failed handling transaction");
+    return true;
+  }
+
+  function raiseAlert(address user) external override {
+      require(address(usersAgent[user]) == msg.sender, "Caller is not agent");
+      agentRaisedAlerts[msg.sender] += 1;
+  } 
+}
+
+contract CryptoVault {
+    address public sweptTokensRecipient;
+    IERC20 public underlying;
+
+    constructor(address recipient) public {
+        sweptTokensRecipient = recipient;
+    }
+
+    function setUnderlying(address latestToken) public {
+        require(address(underlying) == address(0), "Already set");
+        underlying = IERC20(latestToken);
+    }
+
+    //
+    // CryptoVault code
+    //
+
```

Comment:
I'd just remove this, it looks like a title for what follows.

Reply:
This was meant to say that CryptoVault generally has more functionalities than just a "sweepFunction". I would rather chang it for something more explicit like

```
/* ... */ 
```